speech-to-text/src/main/java/com/ibm/watson/developer_cloud/speech_to_text/v1/websocket/WebSocketManager.java

```diff
@@ -144,7 +144,15 @@ public void onMessage(WebSocket webSocket, String message) {
         }
       } else if (json.has(RESULTS) || json.has(SPEAKER_LABELS)) {
         callback.onTranscription(GSON.fromJson(message, SpeechResults.class));
+
       } else if (json.has(STATE)) {
+        // A listen state after everything has been sent over
+        // indicates everything has been processed
+        if (!audioThread.isAlive()) {
+          socket.close(CLOSE_NORMAL, "Transcription completed");
+          return;
+        }
+
         // notify that the service is ready to receive audio
         callback.onListening();
```

Comment:
I don't think we should call `onListening();` if we are going to close the socket.

We should do:
```java
if (!audioThread.isAlive()) {
    socket.close(CLOSE_NORMAL, "Transcription completed");
    return;
} else {
    callback.onListening();
}
```